frontends/tests/run.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env sh
+set -e
+
+# --- Run Appium in background ---
+npx appium --allow-insecure=*:chromedriver_autodownload --port 4723 &
+
+# Wait for Appium to be ready
+until curl -s http://localhost:4723/status | grep '"ready":true'; do
+  echo "Waiting for Appium..."
+  sleep 2
+done
+
+# --- Run tests ---
+npm run test
+TEST_STATUS=$?
+echo "Tests finished with status: $TEST_STATUS"
+
+kill -9 $(pgrep -f appium)
```

Comment:
Why `-9`, that does not give the process a chance to cleanup? 

You can get the PID with `$!` after launching it, see the other comment.